core/src.ts/web/examples/gateway_requestor.html

```diff
@@ -63,23 +74,69 @@ <h2>Using gateway</h2>
     <div id="server-ver"></div>
 
     <script type="text/javascript">
-        // generate random message on load
-        let gate = null;
+        // generate random messages on load
+        generateRandomMessage();
+        generateRandomDigest();
 
-        let rnd = new Uint8Array(6);
-        crypto.getRandomValues(rnd);
-        document.getElementById('message').value = arr2hex(rnd);
-        document.getElementById('signBtn').disabled = true;
-        document.getElementById('signBtn').style.display = 'none';
-
-        let cmd = {"name": "sign", "message": document.getElementById('message').value, "keyNo": 1};
+        let gate = null;
 
         function log(data) {
-            console.log(data);
             document.getElementById('statusText').innerText += '\n' + data;
         }
 
+        function onChangeSelect(selectElement) {
+            const val = selectElement.value;
+            if (val === "message") {
+                document.getElementById('message-container').style.display = 'block';
+                document.getElementById('digest-container').style.display = 'none';
+            } else {
+                document.getElementById('message-container').style.display = 'none';
+                document.getElementById('digest-container').style.display = 'block';
+            }
+        }
+
+        function generateRandomMessage() {
+            const rnd = new Uint8Array(6);
+            crypto.getRandomValues(rnd);
+            document.getElementById('message').value = arr2hex(rnd);
+            document.getElementById('signBtn').disabled = true;
+            document.getElementById('signBtn').style.display = 'none';
+        }
+
+        async function generateRandomDigest() {
+            const rnd = new Uint8Array(6);
+            crypto.getRandomValues(rnd);
+
+            const msgBuffer = new TextEncoder().encode(rnd);     
+            const hashBuffer = await crypto.subtle.digest('SHA-256', msgBuffer);
+            const hashArray = Array.from(new Uint8Array(hashBuffer));
+  
+            const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('');
+
+            document.getElementById('digest').value = hashHex;
+            document.getElementById('signBtn').disabled = true;
+            document.getElementById('signBtn').style.display = 'none';
+        }
```

Comment:
Just a note, any random 32 byte hex string would be a valid SHA-256 hash. Of course then we wouldn't know what was the original input that was hashed, but it's also not necessary for this demo.

Nevertheless, I like the approach of generating a random message and then hashing it on the client side, it's a bit more descriptive. LGTM :+1: